cognition_objects/project.py

```diff
@@ -14,19 +14,51 @@ def get(project_id: str) -> CognitionProject:
     )
 
 
-def get_all(org_id) -> List[CognitionProject]:
+def get_by_user(project_id: str, user_id: str) -> CognitionProject:
     return (
         session.query(CognitionProject)
-        .filter(CognitionProject.organization_id == org_id)
-        .order_by(CognitionProject.created_at.asc())
-        .all()
+        .join(TeamResource, TeamResource.resource_id == CognitionProject.id)
+        .join(TeamMember, TeamMember.team_id == TeamResource.team_id)
+        .filter(TeamMember.user_id == user_id)
+        .filter(
+            TeamResource.resource_type == enums.ResourceType.COGNITION_PROJECT.value
+        )
+        .filter(CognitionProject.id == project_id)
+        .first()
+    )
+
+
+def get_all(org_id: str, order_by_name: bool = False) -> List[CognitionProject]:
+    query = session.query(CognitionProject).filter(
+        CognitionProject.organization_id == org_id
     )
 
+    if order_by_name:
+        query = query.order_by(CognitionProject.name.asc())
+    else:
+        query = query.order_by(CognitionProject.created_at.asc())
+    return query.all()
+
 
 def get_all_all() -> List[CognitionProject]:
     return session.query(CognitionProject).all()
 
 
+def get_all_by_user(org_id: str, user_id: str) -> List[CognitionProject]:
+    return (
+        session.query(CognitionProject)
+        .join(TeamResource, TeamResource.resource_id == CognitionProject.id)
+        .join(TeamMember, TeamMember.team_id == TeamResource.team_id)
+        .filter(TeamMember.user_id == user_id)
+        .filter(
+            TeamResource.resource_type == enums.ResourceType.COGNITION_PROJECT.value
+        )
+        .filter(CognitionProject.organization_id == org_id)
+        .order_by(CognitionProject.created_at.asc())
+        .all()
+    )
+
```

Comment:
i feel like this could cause issues. (Maybe some misconception i have)

i read this as "get me all projects user x has access to". If i understand the logic right this only works if add all engineers  are also part of the team tables.

So maybe a differnt function name or comment to clarify that it's not "get me all projects user x has access to"

Reply:
I see the problem. I decided to move the check for the user role inside the submodule model. So, now `get_all_by_user` and `get_by_user` work for all user roles. When these methods are used, it is checked if the given user has access, considering his role.
So, if the submodule is used somewhere we don't need to write the code for different behaviour for each role again. I feel like this is clearer and safer now. Do you agree?

Reply:
i like it :)